src/components/settings-menu/BackupSection/AlreadyBackedUpView.js

```diff
@@ -0,0 +1,238 @@
+import { sheetVerticalOffset } from '../../../navigation/effects';
+import Routes from '../../../navigation/routesNames';
+import { usePortal } from '../../../react-native-cool-modals/Portal';
+import { setIsWalletLoading, setWalletBackedUp } from '../../../redux/wallets';
+import { colors, fonts, padding } from '../../../styles';
+import { logger } from '../../../utils';
+import { ButtonPressAnimation } from '../../animations';
+import { Centered, Column } from '../../layout';
+import LoadingOverlay, {
+  LoadingOverlayWrapper,
+} from '../../modal/LoadingOverlay';
+import { SheetActionButton } from '../../sheet';
+import { Text } from '../../text';
+
+const DescriptionText = styled(Text).attrs({
+  align: 'center',
+  color: colors.alpha(colors.blueGreyDark, 0.5),
+  lineHeight: 'loosest',
+  size: 'large',
+})`
+  margin-bottom: 42;
+  padding-horizontal: 23;
+`;
+
+const Subtitle = styled(Text).attrs({
+  align: 'center',
+  color: colors.alpha(colors.blueGreyDark, 0.5),
+  size: fonts.size.smedium,
+  weight: fonts.weight.medium,
+})`
+  margin-top: -10;
+`;
+
+const Title = styled(Text).attrs({
+  align: 'center',
+  size: 'larger',
+  weight: 'bold',
+})`
+  margin-bottom: 8;
+  padding-horizontal: 11;
+`;
+
+const TopIcon = styled(View)`
+  border-radius: 25;
+  height: 50;
+  margin-bottom: 19;
+  padding-top: 13;
+  width: 50;
+`;
+
+const TopIconGreen = styled(TopIcon)`
+  background-color: ${colors.green};
+  box-shadow: 0 4px 6px ${colors.alpha(colors.green, 0.4)};
+`;
+
+const TopIconGrey = styled(TopIcon)`
+  background-color: ${colors.blueGreyDark50};
+  box-shadow: 0 4px 6px ${colors.alpha(colors.blueGreyDark50, 0.4)};
+`;
+
+const AlreadyBackedUpView = () => {
+  const { navigate } = useNavigation();
+  const { params } = useRoute();
+  const dispatch = useDispatch();
+  const {
+    isWalletLoading,
+    latestBackup,
+    wallets,
+    selectedWallet,
+  } = useWallets();
+  const walletId = params?.walletId || selectedWallet.id;
+
+  const { setComponent, hide } = usePortal();
+
+  useEffect(() => {
+    if (isWalletLoading) {
+      setComponent(
+        <LoadingOverlayWrapper>
+          <LoadingOverlay
+            paddingTop={sheetVerticalOffset}
+            title={isWalletLoading}
+          />
+        </LoadingOverlayWrapper>,
+        false
+      );
+    }
+    return hide;
+  }, [hide, isWalletLoading, setComponent]);
+
+  const onViewRecoveryPhrase = useCallback(() => {
+    navigate('ShowSecretView', {
+      title: `Recovery ${
+        WalletTypes.mnemonic === wallets[walletId].type ? 'Phrase' : 'Key'
+      }`,
+      walletId,
+    });
+  }, [navigate, walletId, wallets]);
+
+  const walletStatus = useMemo(() => {
+    let status = null;
+    if (wallets[walletId].backedUp) {
+      if (wallets[walletId].backupType === WalletBackupTypes.manual) {
+        status = 'manual_backup';
+      } else {
+        status = 'cloud_backup';
+      }
+    } else {
+      status = 'imported';
```

Comment:
can we extract it somewhere?

Reply:
I mean this status names. We can make a file with these strings.

Reply:
Totally get your point! Just for some context, these strings are used on this file only (and there's no reason to think they would be reused somewhere else in the future) since it's just a name for the state of the view, so I don't think it's worth doing in this case but to make it cleaner I'll move it to constants  at the top of the file.  WDYT?